barretenberg/cpp/src/barretenberg/solidity_helpers/honk_proof_gen.cpp

```diff
@@ -87,19 +87,32 @@ int main(int argc, char** argv)
         inputs[count++] = uint256_t(padded);
     }
 
-    if (plonk_flavour != "honk") {
-        info("Only honk flavor allowed");
-        return 1;
-    }
+    // TODO(https://github.com/AztecProtocol/barretenberg/issues/1227)
+    if (flavor == "honk") {
+        if (circuit_type == "blake") {
+            generate_proof<BlakeCircuit, UltraKeccakFlavor>(inputs);
+        } else if (circuit_type == "add2") {
+            generate_proof<Add2Circuit, UltraKeccakFlavor>(inputs);
+        } else if (circuit_type == "ecdsa") {
+            generate_proof<EcdsaCircuit, UltraKeccakFlavor>(inputs);
+        } else {
+            info("Invalid circuit type: " + circuit_type);
+            return 1;
+        }
 
-    if (circuit_flavour == "blake") {
-        generate_proof<BlakeCircuit>(inputs);
-    } else if (circuit_flavour == "add2") {
-        generate_proof<Add2Circuit>(inputs);
-    } else if (circuit_flavour == "ecdsa") {
-        generate_proof<EcdsaCircuit>(inputs);
+    } else if (flavor == "honk_zk") {
```

Comment:
I know this is messy, working on a follow-up PR that cleans this solidity_helper stuff